core/src/network/reducers/publish.rs

```diff
@@ -43,6 +45,29 @@ fn publish_entry(
     )
 }
 
+/// Send to network a request to publish a header entry alone
+/// This is similar to publishing a regular entry but has limited aspects
+fn publish_header(
+    network_state: &mut NetworkState,
+    chain_header: &ChainHeader,
+) -> Result<(), HolochainError> {
+    let header_entry = Entry::ChainHeader(chain_header.clone());
+    send(
+        network_state,
+        Lib3hClientProtocol::PublishEntry(ProvidedEntryData {
+            space_address: network_state.dna_address.clone().unwrap().into(),
+            provider_agent_id: network_state.agent_id.clone().unwrap().into(),
+            entry: EntryData {
+                entry_address: header_entry.address().clone(),
+                aspect_list: vec![EntryAspect::Header(
```

Comment:
Hah, interesting. So the tests do pass because you've used the header's address as the `entry_address` here, but then it is a bit weird to use the `EntryAspect::Header`, since this data is rather the content of the entry that is the header.